src/plugins/config_reader.go

```diff
@@ -102,38 +141,49 @@ func (r *ConfigReader) updateAgentConfig(cmd *proto.Command) {
 			}
 		}
 
-		for index, feature := range commandData.AgentConfig.Details.Features {
-			commandData.AgentConfig.Details.Features[index] = strings.Replace(feature, "features_", "", 1)
+		if synchronizeFeatures {
+			r.synchronizeFeatures(payloadAgentConfig)
 		}
 
-		sort.Strings(commandData.AgentConfig.Details.Features)
-		sort.Strings(r.config.Features)
-
-		synchronizedFeatures := reflect.DeepEqual(commandData.AgentConfig.Details.Features, r.config.Features)
-
-		if !synchronizedFeatures {
-			for _, feature := range r.config.Features {
-				if feature != agent_config.FeatureRegistration {
-					r.mu.Lock()
-					r.deregisterPlugin(feature)
-					r.mu.Unlock()
-				}
+		r.messagePipeline.Process(core.NewMessage(core.AgentConfigChanged, payloadAgentConfig))
 
-			}
+	} else {
```

Comment:
I think you can remove this else statement as well. If the payloadAgentConfig is nil then we shouldn't be updating the settings

Reply:
Currently if the backoff settings are nil they get set to a default, was trying to keep the same behaviour but can change it if that behaviour was not correct https://github.com/nginx/agent/blob/e59c2a3fdda2175ef15c34c31c293d8df4fe0bbe/src/plugins/commander.go#L86

Reply:
Ok lets keep it as it is for now. I'll address the issue in a separate PR.